tl.tl

```diff
@@ -3514,8 +3523,25 @@ function tl.pretty_print_ast(ast: Node, mode: boolean | PrettyPrintOpts): string
       ["local_declaration"] = {
          after = function(node: Node, children: {Output}): Output
             local out: Output = { y = node.y, h = 0 }
-            table.insert(out, "local")
-            add_child(out, children[1], " ")
+            table.insert(out, "local ")
+            for i, var in ipairs(node.vars) do
+               if i > 1 then
+                  add_string(out, ", ")
+               end
+               add_string(out, var.tk)
+               if var.attribute then
+                  if gen_target ~= "5.4" and var.attribute == "close" then
+                     -- TODO(euclidianAce): I consider this error somewhat appropriate as it is a misuse of the api
+                     -- (also it's a little hard to propagate the error to return nil, err),
+                     -- but it would be better to return a usual error
```

Comment:
You could just let the visitor run all the way, but store the error in a variable at the top-level of the function, and then if the error is set, return nil, err instead of the result.